### PR TITLE
[Backend] NNPI V1.6.0.2 Changes

### PR DESCRIPTION
**Pull request by amitchawla1**
Summary:

1. Initial support for weights sharing.
2. Compilation fix for custom injectors headers files.

Documentation:

1. For module with multiple functions, we disable weights from memory pool. Weights with SHA will be shared across functions.
2. Internal builds doesn't start with 'glow' so path correction for nnpi custom injector header files.

### CODE REVIEW

File: lib/Backends/NNPI/Importer.cpp [hunk 1]
```diff
@@ -912,6 +912,28 @@ class ScatterDataNodeImporter : public INNPINodeImporter {
...
+                          NNPI_INVALID_PARAM);
+
+    importer.setUsedTensors({nodeValueName(glowBucketize->getInput())},
+                            {nodeValueName(glowBucketize->getResult())});
+
+    std::vector<float> bucketizeBoundaries(
+        glowBucketize->getBoundaries().begin(),
+        glowBucketize->getBoundaries().end());
+
+    return nnpiNetworkAddBucketizeOp(
```
Comment (hl475) [inline lib/Backends/NNPI/Importer.cpp:929]: Where is nnpiNetworkAddBucketizeOp defined?

File: lib/Backends/NNPI/Importer.cpp [hunk 2]
```diff
@@ -2624,6 +2646,33 @@ class ResizeNodeImporter : public INNPINodeImporter {
...
+
+    importer.setUsedTensors(
+        {nodeValueName(glowSpLabSplitNode->getLengths()),
+         nodeValueName(glowSpLabSplitNode->getIndices()),
+         nodeValueName(glowSpLabSplitNode->getValues())},
+        {nodeValueName(glowSpLabSplitNode->getLabelValues()),
+         nodeValueName(glowSpLabSplitNode->getExampleIds()),
+         nodeValueName(glowSpLabSplitNode->getGradientOffsetMap())});
+
+    return nnpiNetworkSparseLabelSplitOp(
```
Comment (hl475) [inline lib/Backends/NNPI/Importer.cpp:2665]: Where is nnpiNetworkSparseLabelSplitOp defined?

File: lib/Backends/NNPI/CustomKernels/DSPInjectors/DSPInjectorUtils.cpp [hunk 3]
```diff
@@ -1,6 +1,6 @@
 // (c) Facebook, Inc. and its affiliates. Confidential and proprietary.
 
-#include "glow/lib/Backends/NNPI/CustomKernels/DSPInjectors/DSPInjectorUtils.h"
+#include "DSPInjectorUtils.h"
```
Comment (hl475) [inline lib/Backends/NNPI/CustomKernels/DSPInjectors/DSPInjectorUtils.cpp:3]: Why are you changing this? Does it break the build from your side, or just the lint? And same questions to other places in this PR as well

Comment (amitchawla1) [reply lib/Backends/NNPI/CustomKernels/DSPInjectors/DSPInjectorUtils.cpp:3]: Yes, it breaks build at our end. We don't have 'glow' base folder.

Comment (hl475) [reply lib/Backends/NNPI/CustomKernels/DSPInjectors/DSPInjectorUtils.cpp:3]: Thanks! It was added in https://github.com/pytorch/glow/pull/5596 by @ jackm321 who is currently on PTO. I will check with him once he is back for a proper solution

Comment (hl475) [reply lib/Backends/NNPI/CustomKernels/DSPInjectors/DSPInjectorUtils.cpp:3]: I will take care of these header issues in https://github.com/pytorch/glow/pull/5638 . Take a look and let me know if that fix the build for your side. Thanks!

Comment (amitchawla1) [reply lib/Backends/NNPI/CustomKernels/DSPInjectors/DSPInjectorUtils.cpp:3]: It fixes issues except for the new file "CmpInt32Injector.cpp", for which same path change had to be done.

Comment (amitchawla1) [reply lib/Backends/NNPI/CustomKernels/DSPInjectors/DSPInjectorUtils.cpp:3]: Thanks, the same change works for us.

**Pull request closed without merging**